diff --git a/src/civetweb.c b/src/civetweb.c
index 9c70f19e..f3f735a9 100644
--- a/src/civetweb.c
+++ b/src/civetweb.c
@@ -7815,10 +7815,8 @@ interpret_uri(struct mg_connection *conn, /* in/out: request (must be valid) */
 		}
 
 		if (mg_stat(conn, gz_path, filestat)) {
-			if (filestat) {
-				filestat->is_gzipped = 1;
-				*is_found = 1;
-			}
+			filestat->is_gzipped = 1;
+			*is_found = 1;
 			/* Currently gz files can not be scripts. */
 			return;
 		}
@@ -16021,7 +16019,7 @@ set_ports_option(struct mg_context *phys_ctx)
 			mg_cry_ctx_internal(phys_ctx, "%s", "Out of memory");
 			closesocket(so.sock);
 			so.sock = INVALID_SOCKET;
-			mg_free(ptr);
+			phys_ctx->listening_sockets = ptr;
 			continue;
 		}
 
